File: 002-mockowanie/PHP/src/Service/CreateOrderService.php
<?php
declare(strict_types=1);

namespace App\Service;

use App\Util\EmailInterface;
use \Exception;

/**
 * Class CreateOrderService
 * @package App\Handler
 */
class CreateOrderService
{
    /**
     * @var EmailInterface
     */
    private $email;

    /**
     * CreateOrderService constructor.
     * @param EmailInterface $email
     */
    public function __construct(EmailInterface $email)
    {
        $this->email = $email;
    }


    /**
     * @param string $email
     * @throws Exception
     */
    public function createOrder(string $email): void
    {
        if (!$this->email->send($email)) {
            throw new Exception('Email not send');
        }

        // Other business logic
    }
}

File: 002-mockowanie/Java/src/main/java/lukaszstaniszewski/service/CreateOrderService.java
package lukaszstaniszewski.service;

import lukaszstaniszewski.util.EmailInterface;

/**
 * The type Create order service.
 */
public class CreateOrderService {

    private EmailInterface email;

    /**
     * Instantiates a new Create order service.
     *
     * @param email the email
     */
    public CreateOrderService(EmailInterface email) {
        this.email = email;
    }

    /**
     * Create order.
     *
     * @param emailAddress the email
     * @throws Exception the exception
     */
    public void createOrder(String emailAddress) throws Exception {
        if (!this.email.send(emailAddress)) {
            throw new Exception("Email not send!");
        }

        // Other business logic
    }
}
